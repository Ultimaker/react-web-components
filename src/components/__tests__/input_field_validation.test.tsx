// Copyright (c) 2018 Ultimaker B.V.
import 'jest';
import 'jsdom-global/register';
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import InputFieldValidation from '../input_field_validation';

describe('The InputFieldValidation component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            validationError: 'Validation error',
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm'
        };
        wrapper = shallow(<InputFieldValidation {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should apply the correct positional class', () => {
        wrapper.setProps({ labelLayoutWidth: 'fill' });
        expect(wrapper).toMatchSnapshot();
    });

    it('should apply the correct offset class', () => {
        wrapper.setProps({ labelLayoutWidth: '1/3' });
        expect(wrapper).toMatchSnapshot();
    });

    it('should apply the correct offset class for the required icon', () => {
        wrapper.setProps({ labelLayoutWidth: 'fit', required: true });
        expect(wrapper).toMatchSnapshot();
    });

});
