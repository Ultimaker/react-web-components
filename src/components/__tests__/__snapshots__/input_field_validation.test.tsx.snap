// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The InputFieldValidation component should apply the correct offset class 1`] = `
<div
  className="layout__item u-full input-field__validation"
>
  <div
    className="layout"
  >
    <div
      className="layout__item u-1/3-sm"
    />
    <div
      className="layout__item u-fill "
    >
      <div
        className="input-field__error-message"
      >
        Validation error
      </div>
    </div>
  </div>
</div>
`;

exports[`The InputFieldValidation component should apply the correct offset class for the required icon 1`] = `
<div
  className="layout__item u-full input-field__validation"
>
  <div
    className="layout"
  >
    <div
      className="layout__item u-fill text-right offset-for-required"
    >
      <div
        className="input-field__error-message"
      >
        Validation error
      </div>
    </div>
  </div>
</div>
`;

exports[`The InputFieldValidation component should apply the correct positional class 1`] = `
<div
  className="layout__item u-full input-field__validation"
>
  <div
    className="layout"
  >
    <div
      className="layout__item u-fill text-right"
    >
      <div
        className="input-field__error-message"
      >
        Validation error
      </div>
    </div>
  </div>
</div>
`;

exports[`The InputFieldValidation component should render 1`] = `
<div
  className="layout__item u-full input-field__validation"
>
  <div
    className="layout"
  >
    <div
      className="layout__item u-fill "
    >
      <div
        className="input-field__error-message"
      >
        Validation error
      </div>
    </div>
  </div>
</div>
`;
